### PR TITLE
add getHomeState(), setAway() and setHome()

### PR DESCRIPTION
**Pull request by diplix**
i got this up and running with an older version of pyTado, but this should work, though i didn’t test it. 
to integrate this in homeassistant i made a sensor from the getHomeState() return value and derived a binary sensor from the showHomePresenceSwitchButton attribute.
as soon as showHomePresenceSwitchButton is not none anymore an automation toogles the home state. this mimics the auto assist skill pretty well.
also: i don’t know if or how any of this works if the auto assist skill is active.

### CODE REVIEW

**Comment by diplix** [comment]
i guess this still won’t work after the last commit, because the payload probably has to an array not a json string. right now that’s the best i can do, cause i got the flu, which makes it even harder for me to pretend i’m coding.

File: PyTado/interface.py [hunk 1]
```diff
@@ -355,6 +366,23 @@ def setZoneOverlay(self, zone, overlayMode, setTemp=None, duration=None, deviceT
         data = self._apiCall(cmd, "PUT", post_data)
         return data
 
+    def setHome(self):
+        """Sets HomeState to HOME """
+        # it seems, this can be set anytime
```
Comment (timbru31) [inline PyTado/interface.py]: Nope, the API won't let me do it:

```
{
    "errors": [
        {
            "code": "unableToUpdateHomePresence",
            "title": "Tried to update to AWAY though there is at least one mobile device detected inside the home fence"
        }
    ]
}
```
(It seems the `title` is wrong/swapped (when setting HOME the API says something about AWAY and vice-versa)

**Comment by timbru31** [comment]
FYI (I might open another PR) to set the geolocation of a device one can use the following API call
PUT https://my.tado.com/api/v2/homes/<homeID>/mobileDevices/<mobileDeviceID>/geolocationFix with the following payload. The <mobileDeviceID> can be obtained, e.g., from the /me call:

```
{
    "accuracy": 10,
    "timestamp": "2019-10-29T21:03:21.001Z",
    "geolocation": {
        "longitude": -80.107866,
        "latitude": 39.550314
    }
}
```

**Comment by wmalgadey** [comment]
@diplix and @timbru31 is this working? should we merge ist?

**Comment by diplix** [comment]
sorry, haven’t tested this with this branch/version. i got it working in a much older version, but somebody else should test and confirm this before merging.

**Comment by timbru31** [comment]
I'm using the API calls successfully since ~2 months to toggle my heating from home to away without paying the "Auto Assist", so yes it works. 

**However,** as my comment states, the devices location needs to be in the correct state to toggle.

**Comment by timbru31** [comment]
FYI, whomever is subscribed to this: Tado seems to change this behavior and allow a manual override of the home/away state, regardless of the state of the persons. 👍👍

**Pull request merged**